browser/ui/views/frame/brave_browser_view.cc

```diff
@@ -429,17 +433,114 @@ views::View* BraveBrowserView::GetWalletButtonAnchorView() {
       ->GetAsAnchorView();
 }
 
-std::map<int, std::vector<ui::Accelerator>>
+base::flat_map<int, std::vector<ui::Accelerator>>
 BraveBrowserView::GetAcceleratedCommands() {
   // In future, it will be possible to customize this map so users can configure
   // their own keyboard shortcuts.
-  std::map<int, std::vector<ui::Accelerator>> result;
+  base::flat_map<int, std::vector<ui::Accelerator>> result;
   for (const auto& [accelerator, command] : accelerator_table_) {
     result[command].push_back(accelerator);
   }
   return result;
 }
 
+void BraveBrowserView::AddAcceleratorForCommand(int command_id,
+                                                ui::Accelerator accelerator) {
+  auto replaced = SetAccelerator(command_id, accelerator);
+  // If we replaced a command, make sure we update the prefs.
+  if (replaced) {
+    accelerator_pref_manager_.RemoveAccelerator(replaced.value(), accelerator);
+  }
+  accelerator_pref_manager_.AddAccelerator(command_id, accelerator);
+}
+
+void BraveBrowserView::RemoveAcceleratorForCommand(
+    int command_id,
+    ui::Accelerator accelerator) {
+  auto it = accelerator_table_.find(accelerator);
+  if (it == accelerator_table_.end()) {
+    return;
+  }
+
+  // The accelerator isn't for this command, so don't change it.
+  if (it->second != command_id) {
+    return;
+  }
+
+  auto* focus_manager = GetFocusManager();
+  DCHECK(focus_manager);
+  focus_manager->UnregisterAccelerator(accelerator, this);
+  accelerator_pref_manager_.RemoveAccelerator(command_id, accelerator);
+  accelerator_table_.erase(it);
+}
+
+void BraveBrowserView::ResetAcceleratorsForCommand(int command_id) {
+  auto unsupported = chrome::IsRunningInForcedAppMode() &&
+                     !chrome::IsCommandAllowedInAppMode(
+                         command_id, browser()->is_type_popup());
```

Comment:
Should we add this check into AddAcceleratorForCommand() and RemoveAcceleratorForCommand() too?